src/markdown-pages/install-with-kurl/migrating-csi.md

```diff
@@ -3,73 +3,206 @@ path: "/docs/install-with-kurl/migrating-csi"
-title: "Migrating to Change kURL CSI Add-Ons"
+title: "Migrating to Change CSI Add-On"
 ---
 
-For kURL [v2022.10.28-0](https://kurl.sh/release-notes/v2022.10.28-0) and later, there is a supported data migration path from the Rook CSI add-on to either OpenEBS with Local PV, or Longhorn and MinIO as the new storage provider. 
+This topic describes how to change the Container Storage Interface (CSI) provisioner add-on in your kURL cluster. It includes information about how to use the kURL installer to automatically migrate data to the new provisioner during upgrade. It also includes prerequisites that you must complete before attempting to change CSI add-ons to reduce the risk of errors during data migration.
 
-For information about how to migrate from Rook to a storage provider other than OpenEBS or Longhorn/MinIO, see [Migrating](/docs/install-with-kurl/migrating).
+* [Supported CSI Migrations](#supported-csi-migrations)
+* [About Changing the CSI Add-on](#about-changing-the-csi-add-on)
+  * [Rook to OpenEBS](#rook-to-openebs)
+  * [Longhorn to Rook or OpenEBS](#longhorn-to-rook-or-openebs)
+* [Prerequisites](#prerequisites)
+  * [General Prerequisites](#general-prerequisites)
+  * [Longhorn Prerequisites](#longhorn-prerequisites)
+* [Change the CSI Add-on in a Cluster](#change-the-csi-add-on-in-a-cluster)
+* [Troubleshoot Longhorn Data Migration](#troubleshoot-longhorn-data-migration)  
 
-When initially installing the following kURL spec:
+## Supported CSI Migrations
 
-```
-apiVersion: cluster.kurl.sh/v1beta1
-kind: Installer
-metadata:
-  name: old
-spec:
-  kubernetes:
-    version: 1.19.12
-  docker:
-    version: 20.10.5
-  weave:
-    version: 2.6.5
-  rook:
-    version: 1.0.4
-```
+_**Important**_: kURL does not support Longhorn. If you are currently using Longhorn, you must migrate data from Longhorn to either OpenEBS or Rook.
 
-You can then automatically migrate data _from_ Rook _to_ OpenEBS with Local PV using the following kURL spec. This requires OpenEBS 3.3.0 or newer. 
+This table describes the CSI add-on migration paths that kURL supports:
 
-```
-apiVersion: cluster.kurl.sh/v1beta1
-kind: Installer
-metadata:
-  name: new
-spec:
-  kubernetes:
-    version: 1.19.12
-  docker:
-    version: 20.10.5
-  weave:
-    version: 2.6.5
-  openebs:
-    version: 3.3.0
-    isLocalPVEnabled: true
-```
+| From      | To        | Notes                                                                                                                                                                                                                  |
+|-----------|-----------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Longhorn  | OpenEBS 3.3.0 and later  | Migrating from Longhorn to OpenEBS 3.3.0 or later is recommended for single-node installations. |
```

Comment:
I condensed a lot of the content from the previous draft into the Notes section of the table. Also clarified the supported versions. LMK if anything here is inaccurate or missing; there were lots of details.

Reply:
Checked and everything seems good. Thanks.